flink-core/src/main/java/org/apache/flink/configuration/FallbackKey.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+/**
+ * A key with FallbackKeys will fall back to the FallbackKeys if it itself is not configured.
+ */
+public class FallbackKey {
+
+	private final String key;
+
+	private final boolean isDeprecated;
+
+	public String getKey() {
+		return key;
+	}
+
+	public boolean isDeprecated() {
+		return isDeprecated;
+	}
+
+	FallbackKey(String key, boolean isDeprecated) {
```

Comment:
maybe we could add static methods to create a `FallbackKey`:
```
static FallbackKey createFallbackKey(String key) {
    return new FallbackKey(key, false);
}

static FallbackKey createDeprecatedKey(String key) {
    return new FallbackKey(key, true);
}
```

That way one does not need to remember what the boolean argument does.

Reply:
sound great.